crates/database/src/orders.rs

```diff
@@ -402,6 +402,37 @@ WHERE order_uid = $1
     sqlx::query_as(query).bind(id).fetch_optional(ex).await
 }
 
+pub async fn read_quotes(
+    ex: &mut sqlx::PgConnection,
+    order_ids: &[OrderUid],
+) -> Result<Vec<Quote>, sqlx::Error> {
+    if order_ids.is_empty() {
+        return Ok(vec![]);
+    }
+
+    let placeholders = order_ids
```

Comment:
There is a [query builder](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.size_hint) that seems to accommodate this use case.

As a bonus the builder + taking `order_ids: impl IntoIterator<Item = &OrderUid>` would allow you to create the query without having to allocate a vector for the ids to pass them into the function.
If you still want to have the early return in case the iterator is empty you could do:
```rust
let mut order_ids = order_ids.into_iter().peekable();
if order_ids.peek().is_none() {
    return Ok(vec![]);
}
order_ids
```
Trying to optimize the allocation away is probably a bit overkill but the query builder seems to be reasonable.

Reply:
True, forgot about this before approving the PR. It is already being utilized in the project https://github.com/cowprotocol/services/blob/main/crates/database/src/order_events.rs#L76

Reply:
Ahh, right. This is what I wanted to link: https://docs.rs/sqlx/latest/sqlx/struct.QueryBuilder.html#method.separated

Reply:
refactored. seems a bit nicer now.